wp-content/civi-extensions/goonjcustom/Civi/InstitutionService.php

```diff
@@ -0,0 +1,219 @@
+      $coordinator = $coordinators->itemAt($randomIndex);
+    }
+    else {
+      $coordinator = $coordinators->first();
+    }
+
+    if (!$coordinator) {
+      \CRM_Core_Error::debug_log_message('No coordinator available to assign.');
+      return FALSE;
+    }
+
+    $coordinatorId = $coordinator['contact_id_a'];
+
+    Organization::update(TRUE)
+      ->addValue('Review.Coordinating_POC', $coordinatorId)
+      ->addWhere('id', '=', $contactId)
+      ->execute();
+
+    return TRUE;
+
+  }
+
+  /**
+   * This hook is called after the database write on a custom table.
+   *
+   * @param string $op
+   *   The type of operation being performed.
+   * @param string $objectName
+   *   The custom group ID.
+   * @param int $contactId
+   *   The contactId of the row in the custom table.
+   * @param object $objectRef
+   *   The parameters that were sent into the calling function.
+   */
+
+  /**
+   *
+   */
+
+  /**
+   *
+   */
+  private static function getFallbackOffice() {
+    $fallbackOffices = Contact::get(FALSE)
+      ->addSelect('id')
+      ->addWhere('organization_name', 'CONTAINS', self::FALLBACK_OFFICE_NAME)
+      ->execute();
+
+    return $fallbackOffices->first();
+  }
+
+  /**
+   *
+   */
+  private static function getFallbackCoordinator($contactId) {
+    $fallbackOffice = self::getFallbackOffice();
+
+    // Get the relationship type name based on the institution type.
+    $relationshipTypeName = self::getRelationshipTypeName($contactId);
+
+    $fallbackCoordinators = Relationship::get(FALSE)
+      ->addWhere('contact_id_b', '=', $fallbackOffice['id'])
+      ->addWhere('relationship_type_id:name', '=', $relationshipTypeName)
+      ->addWhere('is_current', '=', TRUE)
+      ->execute();
+
+    $coordinatorCount = $fallbackCoordinators->count();
+
+    $randomIndex = rand(0, $coordinatorCount - 1);
+    $coordinator = $fallbackCoordinators->itemAt($randomIndex);
+
+    return $coordinator;
+  }
+
+  /**
+   *
+   */
+  private static function getRelationshipTypeName($contactId) {
+    $organizations = Organization::get(TRUE)
+      ->addSelect('Institute_Registration.Type_of_Institution:label')
+      ->addWhere('id', '=', $contactId)
+      ->execute()->single();
+
+    $typeOfInstitution = $organizations['Institute_Registration.Type_of_Institution:label'];
+
+    $typeToRelationshipMap = [
+      'Corporate'    => 'Corporate Coordinator of',
+      'Family'       => 'Family Foundation Coordinator of',
+      'Education'    => 'Education Coordinator of',
+      'Government'   => 'Government Coordinator of',
+      'Hospital'     => 'Hospital Coordinator of',
+      'NGO'          => 'NGO Coordinator of',
+      'Others'       => 'Default Coordinator of',
+    ];
+
+    $firstWord = strtok($typeOfInstitution, ' ');
+
+    // Return the corresponding relationship type, or default if not found.
+    return $typeToRelationshipMap[$firstWord] ?? 'Default Coordinator of';
+  }
+
+  /**
+   *
+   */
+
```

Comment:
what's this? Can you please run formatter here

Reply:
@tarunnjoshi done